src/functions.php

```diff
@@ -38,3 +40,32 @@ function usleep(int $time, LoopInterface $loop) : PromiseInterface
 {
     return Timer\resolve($time/1000000, $loop);
 }
+
+/**
+ * Returns the content type in MIME format, like text/plain or application/octet-stream, or FALSE on failure.
+ *
+ * @param string $fileName
+ * @param LoopInterface $loop
+ *
+ * @return PromiseInterface
+ */
+function mime_content_type(string $fileName, LoopInterface $loop) : PromiseInterface
+{
+    $process = new Process(sprintf('file %s --mime-type -b', $fileName));
```

Comment:
Let's use PHP in order to work on all platforms, not only Nix based:
```suggestion
    $process = new Process(sprintf('php -r "echo mime_content_type(\'%s\');"', $fileName));
```